Extension/src/Debugger/configurationProvider.ts

```diff
@@ -453,60 +450,55 @@ export class DebugConfigurationProvider implements vscode.DebugConfigurationProv
-
-                        // Check if debuggerPath exists.
-                        if (await util.checkFileExists(debuggerPath)) {
-                            newConfig.miDebuggerPath = debuggerPath;
-                        } else if ((await util.whichAsync(debuggerName)) !== undefined) {
-                            // Check if debuggerName exists on $PATH
-                            newConfig.miDebuggerPath = debuggerName;
+                        newConfig.type = DebuggerType.cppdbg;
+                    } else if (compilerName === "cl.exe") {
+                        newConfig.miDebuggerPath = undefined;
+                        newConfig.type = DebuggerType.cppvsdbg;
+                        return newConfig;
+                    } else {
+                        debuggerName = "gdb";
+                    }
+                    if (isWindows) {
+                        debuggerName = debuggerName.endsWith(".exe") ? debuggerName : (debuggerName + ".exe");
+                    }
+                    const compilerDirname: string = path.dirname(compilerPath);
+                    const debuggerPath: string = path.join(compilerDirname, debuggerName);
+
+                    // Check if debuggerPath exists.
+                    if (await util.checkFileExists(debuggerPath)) {
+                        newConfig.miDebuggerPath = debuggerPath;
+                    } else if ((await util.whichAsync(debuggerName)) !== undefined) {
+                        // Check if debuggerName exists on $PATH
+                        newConfig.miDebuggerPath = debuggerName;
+                    } else {
+                        // Try the usr path for non-windows platforms.
+                        const usrDebuggerPath: string = path.join("/usr", "bin", debuggerName);
+                        if (!isWindows && await util.checkFileExists(usrDebuggerPath)) {
+                            newConfig.miDebuggerPath = usrDebuggerPath;
                         } else {
-                            // Try the usr path for non-windows platforms.
-                            const usrDebuggerPath: string = path.join("/usr", "bin", debuggerName);
-                            if (!isWindows && await util.checkFileExists(usrDebuggerPath)) {
-                                newConfig.miDebuggerPath = usrDebuggerPath;
-                            } else {
-                                logger.getOutputChannelLogger().appendLine(localize('debugger.path.not.exists', "Unable to find the {0} debugger. The debug configuration for {1} is ignored.", `\"${debuggerName}\"`, compilerName));
-                                return resolve(undefined);
-                            }
+                            logger.getOutputChannelLogger().appendLine(localize('debugger.path.not.exists', "Unable to find the {0} debugger. The debug configuration for {1} is ignored.", `\"${debuggerName}\"`, compilerName));
```

Comment:
Is this supposed to have `void` in front?

Reply:
it's not `async`...